lesson18/script.js

```diff
@@ -0,0 +1,186 @@
+        const slideImage = createAttributedElements({tag:"img",attrObj:{class:"slide-image",src:image.img,alt:image.alt}});
+        
+        fragment.appendChild(sliderItem).appendChild(slideImage);
+    }
+
+    slider.appendChild(fragment);
+}
+
+const getPageNumber = (sliderData) => {
+    const pageNumber    = createAttributedElements({tag:"p",attrObj:{class:"page-number"},str:`/${sliderData.length}`});
+    const currentNumber = createAttributedElements({tag:"span",attrObj:{id:"js-currentNumber",class:"current-number"},str:"1"});
+
+    slider.appendChild(pageNumber).insertAdjacentElement("afterbegin",currentNumber);
+}
+
+const renderButtons = (sliderData) => {
+    const prevBtn = createAttributedElements({tag:"button",attrObj:{id:"js-prevBtn",class:"js-btn prev-btn slider-btn",disabled:true}});
+    const nextBtn = createAttributedElements({tag:"button",attrObj:{id:"js-nextBtn",class:"js-btn next-btn slider-btn"}});
+
+    slider.appendChild(prevBtn).insertAdjacentElement("afterend",nextBtn);
+    addClickEventForButton(sliderData);
+}
+
+const renderPagination = (sliderData) => {
+    const paginationList = createAttributedElements({tag:"ul",attrObj:{id:"js-pagination",class:"pagination"}});
+
+    const fragment = document.createDocumentFragment();
+    const length   = sliderData.length;
+    for(let i = 0; i < length; i++){
+        const paginationItem = createAttributedElements({tag:"li",attrObj:{class:"pagination__item"}});
+        fragment.appendChild(paginationItem);
+    }
+
+    slider.appendChild(paginationList).appendChild(fragment);
+    addClickEventForPagination(sliderData);
+}
+
+let currentIndex = 0;
+const addClickEventForButton = (sliderData) => {
+    const sliderButtons = document.getElementsByClassName("js-btn");
+
+    for (const button of sliderButtons ){
+        button.addEventListener("click",function(){
+            this.id === "js-nextBtn" ? currentIndex++ : currentIndex--;
+
+            switchSlider(sliderData,currentIndex);
+            resetAutoPlay(sliderData);
+        });
+    }
+}
+
+const addClickEventForPagination = (sliderData) => {
+    const paginetions   = document.getElementsByClassName("pagination__item");
+    const arrayPaginations = [...paginetions];
+    
+    for(const pagination of paginetions ){
+        pagination.addEventListener("click",function(){
+            currentIndex = arrayPaginations.indexOf(this);
+
+            switchSlider(sliderData,currentIndex);
+            resetAutoPlay(sliderData);
+        });
+    }
+}
+
+const isActiveFirstItem = () => {
+    document.getElementsByClassName("slider__item")[0].classList.add("is-active");
+    document.getElementsByClassName("pagination__item")[0].classList.add("is-active");
+}
+
+const switchSlider = (sliderData,currentIndex) => {
+    changeCurrentNumber(currentIndex);
+    switchSliderItems(currentIndex);
+    switchPaginations(currentIndex);
```

Comment:
Thank you for your comment!

Absolutely.
The reason for using currentIndex as an argument is that clicking prevBtn or nextBtn after clicking pagination did not work properly.

But this may not have been a problem with the currentIndex.
When I tried without currentIndex, there was no problem, so I fixed it.
https://github.com/nakagawamai/js-lesson/pull/20/commits/9eafa366899763099974960229d319887ad41174